### ISSUE TITLE
Object with '/' in their name are completely not handled on centos 6

### ISSUE THREAD

**Issue by cazino** [scality/ScalitySproxydSwift#120]
Since 3a71d62be2c00bcd5dbae18db5b6e51f9460efc7, the object names are quoted.

Since the 'AllowEncodedSlashes' directive is set to 'Off' (the default value) on centos 6 (because the preferred value 'NoDecode' is not available on that platform), apache respond 404 to every request with a %2F in its url (as expected : http://httpd.apache.org/docs/2.2/mod/core.html#allowencodedslashes).
Therefore on centos 6 the situation regarding object with '/' in their name is worst than before 3a71d62be2c00bcd5dbae18db5b6e51f9460efc7 : these objects are completely not handled.

An exemple : https://37.187.159.67:5443/job/swift-functional-tests/DEVSTACK_BRANCH=stable%2Fjuno,host=centos6-large/142/testReport/junit/test.functional.tests/TestObjectVersioning/test_versioning_check_acl/, and the corresponding object-server log : 
```
object-server: 127.0.0.1 - - [22/May/2015:09:01:21 +0000] "GET /sdb1/389/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-objs/9d6f9aad4bbe479ca8e757ca6486c3f1" 200 5 "GET http://localhost/v1/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-objs/9d6f9aad4bbe479ca8e757ca6486c3f1" "tx046e15667e674179b8706-00555ef061" "proxy-server 13634" 0.0063 "-" 13681
object-server: ERROR __call__ error with PUT /sdb1/261/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-versions/0209d6f9aad4bbe479ca8e757ca6486c3f1/1432285281.17366 : #012Traceback (most recent call last):#012  File "/opt/stack/swift/swift/obj/server.py", line 704, in __call__#012    res = method(req)#012  File "/opt/stack/swift/swift/common/utils.py", line 2486, in wrapped#012    return func(*a, **kw)#012  File "/opt/stack/swift/swift/common/utils.py", line 1060, in _timing_stats#012    resp = func(ctrl, *args, **kwargs)#012  File "/opt/stack/swift/swift/obj/server.py", line 474, in PUT#012    writer.put(metadata)#012  File "/usr/lib/python2.6/site-packages/swift_scality_backend-0.3.2_173_g64e6687-py2.6.egg/swift_scality_backend/utils.py", line 73, in wrapped#012    result = f(*args, **kwargs)#012  File "/usr/lib/python2.6/site-packages/swift_scality_backend-0.3.2_173_g64e6687-py2.6.egg/swift_scality_backend/diskfile.py", line 96, in put#012    str(resp.status), str(msg)))#012SproxydHTTPException: putting: 404 /'&lt;!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN"&gt;#012&lt;html&gt;&lt;head&gt;#012&lt;title&gt;404 Not Found&lt;/title&gt;#012&lt;/head&gt;&lt;body&gt;#012&lt;h1&gt;Not Found&lt;/h1&gt;#012&lt;p&gt;The requested URL /proxy/chord_path/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-versions/0209d6f9aad4bbe479ca8e757ca6486c3f1/1432285281.17366 was not found on this server.&lt;/p&gt;#012&lt;hr&gt;#012&lt;address&gt;Apache/2.2.15 (CentOS) Server at 127.0.0.1 Port 81&lt;/address&gt;#012&lt;/body&gt;&lt;/html&gt;''#012 (txn: tx046e15667e674179b8706-00555ef061)
object-server: 127.0.0.1 - - [22/May/2015:09:01:21 +0000
'&lt;!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN"&gt;#012&lt;html&gt;&lt;head&gt;#012&lt;title&gt;404 Not Found&lt;/title&gt;#012&lt;/head&gt;&lt;body&gt;#012&lt;h1&gt;Not Found&lt;/h1&gt;#012&lt;p&gt;The requested URL /proxy/chord_path/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-versions/0209d6f9aad4bbe479ca8e757ca6486c3f1/1432285281.17366 was not found on this server.&lt;/p&gt;#012&lt;hr&gt;#012&lt;address&gt;Apache/2.2.15 (CentOS) Server at 127.0.0.1 Port 81&lt;/address&gt;#012&lt;/body&gt;&lt;/html&gt;''#012 (txn: tx046e15667e674179b8706-00555ef061)
object-server: 127.0.0.1 - - [22/May/2015:09:01:21 +0000] "PUT /sdb1/261/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-versions/0209d6f9aad4bbe479ca8e757ca6486c3f1/1432285281.17366" 500 1220 "PUT http://localhost/v1/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-versions/0209d6f9aad4bbe479ca8e757ca6486c3f1/1432285281.17366" "tx046e15667e674179b8706-00555ef061" "proxy-server 13634" 0.0070 "-" 13682
object-server: 127.0.0.1 - - [22/May/2015:09:01:21 +0000] "DELETE /sdb1/389/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-objs/9d6f9aad4bbe479ca8e757ca6486c3f1" 204 - "DELETE http://192.168.200.205:8080/v1/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-objs/9d6f9aad4bbe479ca8e757ca6486c3f1" "tx644f823c1bdf450e9820c-00555ef061" "proxy-server 13634" 0.0148 "-" 13682
```

It might not be clear in the above log but after a careful debugging, here is a sum up : 
Sproxyd is asked to perform an operation on this url : /proxy/chord_path/AUTH_454d8144f2ae4c19bb7166460337f5a8/6dd6fb3fcf-versions/0209d6f9aad4bbe479ca8e757ca6486c3f1%2F1432285281.1736 (note the %2F),  apache returns a 404, the object-server gets that 404 and respond 503 to the client.

I think we might want to consider setting  'AllowEncodedSlashes' to 'On' on centos 6 : apache would decode the '/' that the object-server has encoded and we would at least be back to the situation before 3a71d62be2c00bcd5dbae18db5b6e51f9460efc7, which was : 'we don't handle object with leading slashes or back-to-back slashes in their name'.
Note that with this option (but this is already the case), one could not switch from centos6 to ubuntu for running sproxyd (or have a cluster of mixed ubuntu/centos setup) : sproxyd does not decode %2F, so %2F and '/' are two different things for him.

**Comment by JordanP** [comment]
I think setting  'AllowEncodedSlashes' to 'On' on centos 6 is the best option currently. Letter on we might want to upgrade Centos 6 to a recent Apache 2.2 if we find a "decent" package.

**Comment by NicolasT** [comment]
If I'm not mistaken, the proposed solution would extend backwards incompatibility: whilst it's known now that 0.4 will be incompatible with 0.3 objects with `/` in the object name, using the proposed solution on a 0.4-and-up system would result in unretrievable objects when upgrading OS/Apache version. This seems unacceptable.

**Comment by JordanP** [comment]
Closed by #172
